logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/GetLoginFlowForDomainUseCase.kt

```diff
@@ -19,47 +19,102 @@ package com.wire.kalium.logic.feature.auth
 
 import com.wire.kalium.logger.KaliumLogLevel
 import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.configuration.server.CustomServerConfigRepository
+import com.wire.kalium.logic.data.auth.DomainRegistrationMapper
 import com.wire.kalium.logic.data.auth.LoginDomainPath
 import com.wire.kalium.logic.data.auth.login.LoginRepository
-import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.di.MapperProvider
 import com.wire.kalium.logic.functional.fold
-import com.wire.kalium.logic.functional.left
-import com.wire.kalium.logic.functional.right
 import com.wire.kalium.logic.kaliumLogger
 import com.wire.kalium.logic.logStructuredJson
 
 /**
  * Use case to get the login flow for the client app/user to follow.
  * This is determined by the backend according to registration of the domain.¬
  *
- * @param email the email to look up the domain registration for
+ * @param [email] the email to look up the domain registration for
  */
 interface GetLoginFlowForDomainUseCase {
-    suspend operator fun invoke(email: String): Either<CoreFailure, LoginDomainPath>
+    suspend operator fun invoke(email: String): EnterpriseLoginResult
 }
 
 @Suppress("FunctionNaming")
 internal fun GetLoginFlowForDomainUseCase(
-    loginRepository: LoginRepository
+    loginRepository: LoginRepository,
+    customServerConfigRepository: CustomServerConfigRepository,
+    mapper: DomainRegistrationMapper = MapperProvider.domainRegistrationMapper()
 ) = object : GetLoginFlowForDomainUseCase {
-    override suspend fun invoke(email: String): Either<CoreFailure, LoginDomainPath> {
+    override suspend fun invoke(email: String): EnterpriseLoginResult {
         logger.d("Get domain registration")
         return loginRepository.getDomainRegistration(email).fold({
             logger.logStructuredJson(
                 level = KaliumLogLevel.ERROR,
                 leadingMessage = "Get domain registration",
-                jsonStringKeyValues = mapOf("error" to it)
+                jsonStringKeyValues = mapOf("error" to it::class.simpleName)
```

Comment:
On release build where classes are obfuscated, this will just print some symbols